grades/models.py

```diff
@@ -9,30 +9,44 @@ class Course (models.Model):
     credinitials = models.IntegerField(default=2)
 
 
+class Semester(models.TextChoices):
+    Winter = 'A',
+    Spring = 'B',
+    Summer = 'C'
+
+
 class Grade(models.Model):
-    SEMESTER_NAME = (
-        ('A', 'Winter'),
-        ('B', 'Spring'),
-        ('C', 'Summer'),
-    )
-    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE,
-                             related_name='%(class)s_student', null=True, blank=True)
-    course = models.ForeignKey(Course, on_delete=models.RESTRICT, null=True, blank=True)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    course = models.ForeignKey(Course, on_delete=models.RESTRICT)
     grade = models.IntegerField(default=0)
-    semester = models.CharField(max_length=1, choices=SEMESTER_NAME)
+    semester = models.CharField(max_length=1, choices=Semester.choices)
     year = models.IntegerField(default=0)
 
+    @classmethod
+    def addCourse(cls, course_name, credinitials):
+        newCourse = Course()
+        newCourse.course_name = course_name
+        newCourse.credinitials = credinitials
+        newCourse.save()
+        return newCourse
+
+    @classmethod
+    def removeCourse(cls, courseID):
```

Comment:
This should be a classmethod of course, not of grade.

Reply:
Did you meant that it supposed to be under the class Course? @simzacks

Reply:
same as above